tests/test_packages/test_skills/test_generic_seller/test_behaviours.py

```diff
@@ -0,0 +1,222 @@
+        )
+        assert has_attributes, error_str
+        mock_logger.assert_any_call(logging.INFO, "registering agent on SOEF.")
+
+        # message 3
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=OefSearchMessage,
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            to=self.skill.skill_context.search_service_address,
+            sender=self.skill.skill_context.agent_address,
+            service_description=mocked_description_2,
+        )
+        assert has_attributes, error_str
+        mock_logger.assert_any_call(logging.INFO, "registering service on SOEF.")
+
+    def test_service_registration_behaviour_setup_not_is_ledger_tx(self):
+        """Test the setup method of the service_registration behaviour: where is_ledger_tx is False."""
+        # setup
+        self.strategy._is_ledger_tx = False
+        mocked_description_1 = "some_description_1"
+        mocked_description_2 = "some_description_2"
+
+        # operation
+        with patch.object(
+            self.strategy,
+            "get_location_description",
+            return_value=mocked_description_1,
+        ):
+            with patch.object(
+                self.strategy,
+                "get_register_service_description",
+                return_value=mocked_description_2,
+            ):
+                with patch.object(
+                    self.service_registration.context.logger, "log"
+                ) as mock_logger:
+                    self.service_registration.setup()
+
+        # after
+        quantity = self.get_quantity_in_outbox()
+        assert (
+            quantity == 2
+        ), f"Invalid number of messages in outbox. Expected 2. Found {quantity}."
+
+        # message 1
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=OefSearchMessage,
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            to=self.skill.skill_context.search_service_address,
+            sender=self.skill.skill_context.agent_address,
+            service_description=mocked_description_1,
+        )
+        assert has_attributes, error_str
+        mock_logger.assert_any_call(logging.INFO, "registering agent on SOEF.")
+
+        # message 2
+        has_attributes, error_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=OefSearchMessage,
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            to=self.skill.skill_context.search_service_address,
+            sender=self.skill.skill_context.agent_address,
+            service_description=mocked_description_2,
+        )
+        assert has_attributes, error_str
+        mock_logger.assert_any_call(logging.INFO, "registering service on SOEF.")
```

Comment:
In tests you can also use private helper methods to reduce code duplication

Reply:
Done